[release] v5.14.9 (#38920)
mui/material-ui: CHANGELOG.md

# [Versions](https://mui.com/versions/)

## 5.14.9

<!-- generated comparing v5.14.8..master -->

> _Sep 11, 2023_

Reviewer: ```suggestion
_Sep 12, 2023_
```

# [Versions](https://mui.com/versions/)

## 5.14.9

<!-- generated comparing v5.14.8..master -->

_Sep 11, 2023_

A big thanks to the 15 contributors who made this release possible. Here are some highlights ✨:

- 🎉 Added `Drawer` component to Joy UI (#38169) @mnajdova

> - ✨ Added `firstButton`, `lastButton`, and `middleButton` classes to `ButtonGroup`'s button descendants in Material UI (#38520) @ZeeshanTamboli

Reviewer: We can highlight it as:
```suggestion
- ✨ ButtonGroup now styles different button elements within it correctly (#38520) @ZeeshanTamboli
```

## 5.14.8

<!-- generated comparing v5.14.8..master -->

_Sep 11, 2023_

A big thanks to the 15 contributors who made this release possible. Here are some highlights ✨:

> - 🎉 Added `Drawer` component to Joy UI (#38169) @mnajdova

Reviewer 2: A link to the docs could be great.

Author: Added link to the docs
@siriwatknp @zanivan, let me know if you have an image or video you wish to add 😊

Reviewer 3: I don't agree here about images or video clips. We can brag about new features on social media, but the changelog should be concise and highlight changes from the previous version. It's not a means of marketing.

A link is fine, though.

Reviewer 4: Yep, link is enough for the changelog.

Reviewer 2: > I don't agree here about images or video clips. We can brag about new features on social media, but the changelog should be concise and highlight changes from the previous version. It's not a means of marketing.

@michaldudak There are different schools on this.

1. Changelog first: https://github.com/prisma/prisma/releases, https://github.com/drizzle-team/drizzle-orm/releases, 
2. Twitter first: https://github.com/chakra-ui/chakra-ui/releases https://github.com/shadcn-ui/ui/releases
3. Blog posts first: https://github.com/microsoft/TypeScript/releases, https://nextjs.org/blog, https://v5.mantine.dev/changelog/5-8-0/
4. Doesn't announce

I feel the problem in our case is that we fall most often under "Doesn't announce" 😁. I would be much in favor of 3.

> - ✨ Added `firstButton`, `lastButton`, and `middleButton` classes to `ButtonGroup`'s button descendants in Material UI (#38520) @ZeeshanTamboli

Reviewer 2: This change is unclear to me, what's the problem solved? Or maybe it's not worth a highlight.

Author: This is a quality-of-life improvement for users of the `ButtonGroup` component, as they can now target the buttons using classes.

What do you think about

```suggestion
- ✨ The Material UI `ButtonGroup` now provides utility classes for its buttons' position (#38520) @ZeeshanTamboli
```

@ZeeshanTamboli I took inspiration from your suggestion above, but I would like to highlight it positively ("We're adding this quality-of-life enhancement") instead of negatively ("This is now working correctly")

Reviewer: The main aim for the PR was not provide these classes for buttons, but to fix the styling for:
- buttons with hrefs - #31210
- disabled buttons displaying tooltips - #29514

Looking from a user's perspective, I wouldn't be particularly enthusiastic about classes available (especially in a highlight), but on understanding what is the problem solved using those classes. Yes, you can use the classes for customization but that was not the goal of the PR.

Author: @ZeeshanTamboli ok, that makes sense, I updated it 🙌🏼

Reviewer 2: I think we could make the case that #38520 is a breaking change.

### `@mui/material@5.14.9`

> - &#8203;<!-- 44 -->[ButtonGroup][material-ui] Determine first, last and middle buttons to support different elements with correct styling (#38520) @ZeeshanTamboli

Reviewer 2: Maybe we shouldn't remove the product name when it's already under the header.

```suggestion
- &#8203;<!-- 44 -->[ButtonGroup] Determine first, last and middle buttons to support different elements with correct styling (#38520) @ZeeshanTamboli
```

x the other pages.

Author: @mnajdova what do you think?

Reviewer 4: I think it should be removed.

Reviewer 5: Yep, let's remove it.

Author: I removed the product names when the change is already under the header 👍🏼 
Should we do the same for the Docs section (remove `[docs]`) and Core section (remove `[core]`)? 🤔

Reviewer 5: Let's keep it as is now, we can decide this elsewhere so that we don't block this PR. As far as I could see the original point was around Material UI, which is already resolved.
